### DISCUSSION TITLE
How to index a column using Drift?

### DISCUSSION THREAD

**Discussion by TinhHuynh** [simolus3/drift#2041] [Q&A] [answered]
Hi, we have tables generated by Drift. But now we want to index some columns for faster DB queries. 
How do we do that?

**Comment by simolus3** [comment, marked as answer]
Hi, at the moment this is not directly possible with a high-level Dart API. You can write the `CREATE INDEX` statement manually and add it to your database like this:

```dart
  @override
  List<DatabaseSchemaEntity> get allSchemaEntities => [
        ...super.allSchemaEntities,
        _myIndex,
      ];

  final Index _myIndex =
      Index('my_index', 'CREATE INDEX my_index ON foo (bar, baz)');

```

Added like this, the index will be created automatically when the database is created. You can also create it manually with `migrator.create(_myIndex)` in a schema migration.
